### PR TITLE
[CH][KPI] Migrate % of red commits on trunk metric

### PR DESCRIPTION
**Pull request by huydhn**
https://torchci-git-fork-huydhn-ch-mastercommitredpercent-fbopensource.vercel.app/kpis

### CODE REVIEW

File: torchci/clickhouse_queries/master_commit_red_percent/query.sql [hunk 1]
```diff
@@ -80,24 +80,28 @@ any_red AS (
...
     GROUP BY
         granularity_bucket
 ),
 avg_red AS (
     SELECT
-        classified_red.granularity_bucket,
-        ELEMENT_AT(metrics.metric, 1) AS name,
-        ELEMENT_AT(metrics.metric, 2) AS metric,
+        granularity_bucket,
+        arrayElement(metrics, 1) AS name,
```
Comment (clee2000) [inline torchci/clickhouse_queries/master_commit_red_percent/query.sql:101]: minor: I'm pretty sure you can use metrics[1] for this

File: torchci/clickhouse_queries/master_commit_red_percent/query.sql [hunk 2]
```diff
@@ -80,24 +80,28 @@ any_red AS (
...
+        arrayJoin(
+            array(
+                array('Broken trunk', toString(AVG(broken_trunk_red))),
+                array(
+                    'Flaky',
+                    toString(AVG(all_red) - AVG(broken_trunk_red))
+                ),
+                array('Total', toString(AVG(all_red)))
+            )
+        ) AS metrics
```
Comment (clee2000) [inline torchci/clickhouse_queries/master_commit_red_percent/query.sql:92]: Curious as to why you cast to string for all of these?

**Review by clee2000 — APPROVED** [review]
I think there are things that could be done to make this faster, but idk if you care about that, especially since the original RS query also seems slow too

File: torchci/clickhouse_queries/master_commit_red_percent/query.sql (hunk 2 above)
Comment (huydhn) [reply torchci/clickhouse_queries/master_commit_red_percent/query.sql:92]: Oh, I learn that CH only accepts array of the same type, i.e. Array(String), while Rockset, like Python, allows different types to mix together (string and float).  We could have a TODO task here to see if this query could be rewritten in a better way.

**Comment by huydhn** [comment]
> I think there are things that could be done to make this faster, but idk if you care about that, especially since the original RS query also seems slow too

Yeah, that could be a good BE task.  The other issue I have with this one is that it uses too much memory.  An useful skill I want to learn more about is how to optimize the query for memory usage.

**Pull request merged**